[2021-04-24 11:45:10,351] {taskinstance.py:877} INFO - Dependencies all met for <TaskInstance: project_dag.Stage_songs 2020-06-01T00:00:00+00:00 [queued]>
[2021-04-24 11:45:11,257] {taskinstance.py:877} INFO - Dependencies all met for <TaskInstance: project_dag.Stage_songs 2020-06-01T00:00:00+00:00 [queued]>
[2021-04-24 11:45:11,258] {taskinstance.py:1068} INFO - 
--------------------------------------------------------------------------------
[2021-04-24 11:45:11,258] {taskinstance.py:1069} INFO - Starting attempt 1 of 2
[2021-04-24 11:45:11,265] {taskinstance.py:1070} INFO - 
--------------------------------------------------------------------------------
[2021-04-24 11:45:12,138] {taskinstance.py:1089} INFO - Executing <Task(StageToRedshiftOperator): Stage_songs> on 2020-06-01T00:00:00+00:00
[2021-04-24 11:45:12,223] {standard_task_runner.py:52} INFO - Started process 195 to run task
[2021-04-24 11:45:12,451] {standard_task_runner.py:76} INFO - Running: ['airflow', 'tasks', 'run', 'project_dag', 'Stage_songs', '2020-06-01T00:00:00+00:00', '--job-id', '78', '--pool', 'default_pool', '--raw', '--subdir', 'DAGS_FOLDER/udac_example_dag.py', '--cfg-path', '/tmp/tmps_hac54w', '--error-file', '/tmp/tmplex7hnmp']
[2021-04-24 11:45:12,504] {standard_task_runner.py:77} INFO - Job 78: Subtask Stage_songs
[2021-04-24 11:45:14,125] {logging_mixin.py:104} INFO - Running <TaskInstance: project_dag.Stage_songs 2020-06-01T00:00:00+00:00 [running]> on host 88dcd2a12842
[2021-04-24 11:45:15,857] {taskinstance.py:1283} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_EMAIL=ernest@gmail.com.com
AIRFLOW_CTX_DAG_OWNER=udacity
AIRFLOW_CTX_DAG_ID=project_dag
AIRFLOW_CTX_TASK_ID=Stage_songs
AIRFLOW_CTX_EXECUTION_DATE=2020-06-01T00:00:00+00:00
AIRFLOW_CTX_DAG_RUN_ID=scheduled__2020-06-01T00:00:00+00:00
[2021-04-24 11:45:15,873] {stage_redshift.py:38} INFO - Start implementation of StageToRedshiftOperator
[2021-04-24 11:45:16,113] {taskinstance.py:1482} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/models/taskinstance.py", line 1138, in _run_raw_task
    self._prepare_and_execute_task_with_callbacks(context, task)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/models/taskinstance.py", line 1311, in _prepare_and_execute_task_with_callbacks
    result = self._execute_task(context, task_copy)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/models/taskinstance.py", line 1341, in _execute_task
    result = task_copy.execute(context=context)
  File "/opt/airflow/plugins/operators/stage_redshift.py", line 39, in execute
    aws_hook = AwsHook(self.aws_credentials_id)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/contrib/hooks/aws_hook.py", line 41, in __init__
    super().__init__(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/providers/amazon/aws/hooks/base_aws.py", line 360, in __init__
    raise AirflowException('Either client_type or resource_type must be provided.')
airflow.exceptions.AirflowException: Either client_type or resource_type must be provided.
[2021-04-24 11:45:16,154] {taskinstance.py:1532} INFO - Marking task as UP_FOR_RETRY. dag_id=project_dag, task_id=Stage_songs, execution_date=20200601T000000, start_date=20210424T114510, end_date=20210424T114516
[2021-04-24 11:45:16,522] {local_task_job.py:146} INFO - Task exited with return code 1
[2021-04-26 09:07:48,602] {taskinstance.py:877} INFO - Dependencies all met for <TaskInstance: project_dag.Stage_songs 2020-06-01T00:00:00+00:00 [queued]>
[2021-04-26 09:07:49,972] {taskinstance.py:877} INFO - Dependencies all met for <TaskInstance: project_dag.Stage_songs 2020-06-01T00:00:00+00:00 [queued]>
[2021-04-26 09:07:49,973] {taskinstance.py:1068} INFO - 
--------------------------------------------------------------------------------
[2021-04-26 09:07:49,973] {taskinstance.py:1069} INFO - Starting attempt 1 of 2
[2021-04-26 09:07:49,973] {taskinstance.py:1070} INFO - 
--------------------------------------------------------------------------------
[2021-04-26 09:07:50,772] {taskinstance.py:1089} INFO - Executing <Task(StageToRedshiftOperator): Stage_songs> on 2020-06-01T00:00:00+00:00
[2021-04-26 09:07:51,020] {standard_task_runner.py:76} INFO - Running: ['airflow', 'tasks', 'run', 'project_dag', 'Stage_songs', '2020-06-01T00:00:00+00:00', '--job-id', '15', '--pool', 'default_pool', '--raw', '--subdir', 'DAGS_FOLDER/udac_example_dag.py', '--cfg-path', '/tmp/tmp0rn60y96', '--error-file', '/tmp/tmp2vp7sy3b']
[2021-04-26 09:07:50,906] {standard_task_runner.py:52} INFO - Started process 80 to run task
[2021-04-26 09:07:51,022] {standard_task_runner.py:77} INFO - Job 15: Subtask Stage_songs
[2021-04-26 09:07:53,737] {logging_mixin.py:104} INFO - Running <TaskInstance: project_dag.Stage_songs 2020-06-01T00:00:00+00:00 [running]> on host b5c40a6f08c7
[2021-04-26 09:07:54,510] {taskinstance.py:1283} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_EMAIL=ernestoffor@gmail.com.com
AIRFLOW_CTX_DAG_OWNER=udacity
AIRFLOW_CTX_DAG_ID=project_dag
AIRFLOW_CTX_TASK_ID=Stage_songs
AIRFLOW_CTX_EXECUTION_DATE=2020-06-01T00:00:00+00:00
AIRFLOW_CTX_DAG_RUN_ID=scheduled__2020-06-01T00:00:00+00:00
[2021-04-26 09:07:54,511] {stage_redshift.py:63} INFO - Start implementation of StageToRedshiftOperator
[2021-04-26 09:07:54,524] {stage_redshift.py:68} INFO - Preparing destination Redshift table
[2021-04-26 09:07:54,637] {base.py:78} INFO - Using connection to: id: redshift. Host: redshift.cxvwoeu3kuea.us-west-2.redshift.amazonaws.com, Port: 5439, Schema: dev, Login: awsuser, Password: XXXXXXXX, extra: None
[2021-04-26 09:07:56,153] {dbapi.py:180} INFO - Running statement: DELETE FROM staging_songs, parameters: None
[2021-04-26 09:07:56,660] {dbapi.py:186} INFO - Rows affected: 0
[2021-04-26 09:07:57,029] {stage_redshift.py:71} INFO - Getting data from S3 to Redshift
[2021-04-26 09:07:57,107] {base.py:78} INFO - Using connection to: id: redshift. Host: redshift.cxvwoeu3kuea.us-west-2.redshift.amazonaws.com, Port: 5439, Schema: dev, Login: awsuser, Password: XXXXXXXX, extra: None
[2021-04-26 09:07:58,564] {dbapi.py:180} INFO - Running statement: 
             COPY staging_songs 
             FROM 's3://udacity-dend/song_data/*/*/*/'
             FORMAT AS JSON 'JSON 'auto''
             CREDENTIALS
             'aws_access_key_id=None;aws_secret_access_key=None'
                , parameters: None
[2021-04-26 09:07:58,980] {taskinstance.py:1482} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/models/taskinstance.py", line 1138, in _run_raw_task
    self._prepare_and_execute_task_with_callbacks(context, task)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/models/taskinstance.py", line 1311, in _prepare_and_execute_task_with_callbacks
    result = self._execute_task(context, task_copy)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/models/taskinstance.py", line 1341, in _execute_task
    result = task_copy.execute(context=context)
  File "/opt/airflow/plugins/operators/stage_redshift.py", line 81, in execute
    redshift.run(execution_sql)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/hooks/dbapi.py", line 184, in run
    cur.execute(sql_statement)
psycopg2.errors.SyntaxError: syntax error at or near "auto"
LINE 4:              FORMAT AS JSON 'JSON 'auto''
                                           ^

[2021-04-26 09:07:59,061] {taskinstance.py:1532} INFO - Marking task as UP_FOR_RETRY. dag_id=project_dag, task_id=Stage_songs, execution_date=20200601T000000, start_date=20210426T090748, end_date=20210426T090759
[2021-04-26 09:07:59,947] {local_task_job.py:146} INFO - Task exited with return code 1
[2021-04-26 12:34:58,739] {taskinstance.py:877} INFO - Dependencies all met for <TaskInstance: project_dag.Stage_songs 2020-06-01T00:00:00+00:00 [queued]>
[2021-04-26 12:34:58,890] {taskinstance.py:877} INFO - Dependencies all met for <TaskInstance: project_dag.Stage_songs 2020-06-01T00:00:00+00:00 [queued]>
[2021-04-26 12:34:58,891] {taskinstance.py:1068} INFO - 
--------------------------------------------------------------------------------
[2021-04-26 12:34:58,891] {taskinstance.py:1069} INFO - Starting attempt 1 of 2
[2021-04-26 12:34:58,891] {taskinstance.py:1070} INFO - 
--------------------------------------------------------------------------------
[2021-04-26 12:34:58,995] {taskinstance.py:1089} INFO - Executing <Task(StageToRedshiftOperator): Stage_songs> on 2020-06-01T00:00:00+00:00
[2021-04-26 12:34:59,061] {standard_task_runner.py:76} INFO - Running: ['airflow', 'tasks', 'run', 'project_dag', 'Stage_songs', '2020-06-01T00:00:00+00:00', '--job-id', '9', '--pool', 'default_pool', '--raw', '--subdir', 'DAGS_FOLDER/udac_example_dag.py', '--cfg-path', '/tmp/tmph3sr2h7i', '--error-file', '/tmp/tmpi440gmng']
[2021-04-26 12:34:59,062] {standard_task_runner.py:77} INFO - Job 9: Subtask Stage_songs
[2021-04-26 12:34:59,015] {standard_task_runner.py:52} INFO - Started process 68 to run task
[2021-04-26 12:34:59,595] {logging_mixin.py:104} INFO - Running <TaskInstance: project_dag.Stage_songs 2020-06-01T00:00:00+00:00 [running]> on host 0f1e1da1f4f6
[2021-04-26 12:35:00,187] {taskinstance.py:1283} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_EMAIL=ernestoffor@gmail.com.com
AIRFLOW_CTX_DAG_OWNER=udacity
AIRFLOW_CTX_DAG_ID=project_dag
AIRFLOW_CTX_TASK_ID=Stage_songs
AIRFLOW_CTX_EXECUTION_DATE=2020-06-01T00:00:00+00:00
AIRFLOW_CTX_DAG_RUN_ID=scheduled__2020-06-01T00:00:00+00:00
[2021-04-26 12:35:00,188] {stage_redshift.py:62} INFO - Start implementation of StageToRedshiftOperator
[2021-04-26 12:35:00,189] {stage_redshift.py:67} INFO - Preparing destination Redshift table
[2021-04-26 12:35:00,318] {base.py:78} INFO - Using connection to: id: redshift. Host: redshift.cxvwoeu3kuea.us-west-2.redshift.amazonaws.com, Port: 5439, Schema: dev, Login: awsuser, Password: XXXXXXXX, extra: None
[2021-04-26 12:35:01,824] {dbapi.py:180} INFO - Running statement: DELETE FROM staging_songs, parameters: None
[2021-04-26 12:35:02,733] {dbapi.py:186} INFO - Rows affected: 0
[2021-04-26 12:35:03,058] {stage_redshift.py:70} INFO - Getting data from S3 to Redshift
[2021-04-26 12:35:03,117] {base.py:78} INFO - Using connection to: id: redshift. Host: redshift.cxvwoeu3kuea.us-west-2.redshift.amazonaws.com, Port: 5439, Schema: dev, Login: awsuser, Password: XXXXXXXX, extra: None
[2021-04-26 12:35:04,517] {dbapi.py:180} INFO - Running statement: 
             COPY staging_songs 
             FROM '/offor_data/song_data/*/*/*/'
             FORMAT AS JSON 'JSON 'auto''
                  , parameters: None
[2021-04-26 12:35:04,905] {taskinstance.py:1482} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/models/taskinstance.py", line 1138, in _run_raw_task
    self._prepare_and_execute_task_with_callbacks(context, task)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/models/taskinstance.py", line 1311, in _prepare_and_execute_task_with_callbacks
    result = self._execute_task(context, task_copy)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/models/taskinstance.py", line 1341, in _execute_task
    result = task_copy.execute(context=context)
  File "/opt/airflow/plugins/operators/stage_redshift.py", line 80, in execute
    redshift.run(execution_sql)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/hooks/dbapi.py", line 184, in run
    cur.execute(sql_statement)
psycopg2.errors.SyntaxError: syntax error at or near "auto"
LINE 4:              FORMAT AS JSON 'JSON 'auto''
                                           ^

[2021-04-26 12:35:04,911] {taskinstance.py:1532} INFO - Marking task as UP_FOR_RETRY. dag_id=project_dag, task_id=Stage_songs, execution_date=20200601T000000, start_date=20210426T123458, end_date=20210426T123504
[2021-04-26 12:35:05,050] {local_task_job.py:146} INFO - Task exited with return code 1
[2021-04-27 20:39:02,764] {taskinstance.py:877} INFO - Dependencies all met for <TaskInstance: project_dag.Stage_songs 2020-06-01T00:00:00+00:00 [queued]>
[2021-04-27 20:39:03,262] {taskinstance.py:877} INFO - Dependencies all met for <TaskInstance: project_dag.Stage_songs 2020-06-01T00:00:00+00:00 [queued]>
[2021-04-27 20:39:03,274] {taskinstance.py:1068} INFO - 
--------------------------------------------------------------------------------
[2021-04-27 20:39:03,283] {taskinstance.py:1069} INFO - Starting attempt 1 of 2
[2021-04-27 20:39:03,283] {taskinstance.py:1070} INFO - 
--------------------------------------------------------------------------------
[2021-04-27 20:39:03,550] {taskinstance.py:1089} INFO - Executing <Task(StageToRedshiftOperator): Stage_songs> on 2020-06-01T00:00:00+00:00
[2021-04-27 20:39:03,854] {standard_task_runner.py:76} INFO - Running: ['airflow', 'tasks', 'run', 'project_dag', 'Stage_songs', '2020-06-01T00:00:00+00:00', '--job-id', '69', '--pool', 'default_pool', '--raw', '--subdir', 'DAGS_FOLDER/udac_example_dag.py', '--cfg-path', '/tmp/tmpjw7eap7x', '--error-file', '/tmp/tmpov419fvm']
[2021-04-27 20:39:03,728] {standard_task_runner.py:52} INFO - Started process 182 to run task
[2021-04-27 20:39:03,964] {standard_task_runner.py:77} INFO - Job 69: Subtask Stage_songs
[2021-04-27 20:39:05,423] {logging_mixin.py:104} INFO - Running <TaskInstance: project_dag.Stage_songs 2020-06-01T00:00:00+00:00 [running]> on host 478c7ec349b0
[2021-04-27 20:39:07,066] {taskinstance.py:1283} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_EMAIL=ernestoffor@gmail.com.com
AIRFLOW_CTX_DAG_OWNER=udacity
AIRFLOW_CTX_DAG_ID=project_dag
AIRFLOW_CTX_TASK_ID=Stage_songs
AIRFLOW_CTX_EXECUTION_DATE=2020-06-01T00:00:00+00:00
AIRFLOW_CTX_DAG_RUN_ID=scheduled__2020-06-01T00:00:00+00:00
[2021-04-27 20:39:07,067] {stage_redshift.py:67} INFO - Start implementation of StageToRedshiftOperator
[2021-04-27 20:39:07,144] {stage_redshift.py:74} INFO - Preparing destination Redshift table
[2021-04-27 20:39:07,596] {base.py:78} INFO - Using connection to: id: redshift. Host: redshift.cxvwoeu3kuea.us-west-2.redshift.amazonaws.com, Port: 5439, Schema: dev, Login: awsuser, Password: XXXXXXXX, extra: None
[2021-04-27 20:39:09,093] {dbapi.py:180} INFO - Running statement: DELETE FROM staging_songs, parameters: None
[2021-04-27 20:39:09,537] {dbapi.py:186} INFO - Rows affected: 0
[2021-04-27 20:39:09,860] {stage_redshift.py:77} INFO - Getting data from S3 to Redshift
[2021-04-27 20:39:14,296] {taskinstance.py:1482} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/models/taskinstance.py", line 1138, in _run_raw_task
    self._prepare_and_execute_task_with_callbacks(context, task)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/models/taskinstance.py", line 1311, in _prepare_and_execute_task_with_callbacks
    result = self._execute_task(context, task_copy)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/models/taskinstance.py", line 1341, in _execute_task
    result = task_copy.execute(context=context)
  File "/opt/airflow/plugins/operators/stage_redshift.py", line 83, in execute
    s3.get_credentials().access_key,
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/providers/amazon/aws/hooks/base_aws.py", line 478, in get_credentials
    return session.get_credentials().get_frozen_credentials()
AttributeError: 'NoneType' object has no attribute 'get_frozen_credentials'
[2021-04-27 20:39:14,341] {taskinstance.py:1532} INFO - Marking task as UP_FOR_RETRY. dag_id=project_dag, task_id=Stage_songs, execution_date=20200601T000000, start_date=20210427T203902, end_date=20210427T203914
[2021-04-27 20:39:15,662] {local_task_job.py:146} INFO - Task exited with return code 1
